File: src/week2/report2/Screening.java
package week2.report2;

import java.time.LocalDateTime;

public class Screening {
  
  final int sequence;
  final LocalDateTime whenScreened;
    
  public Screening(int sequence, LocalDateTime when) {
    this.sequence = sequence;
    this.whenScreened = when;
    
  }
  
  
  /*boolean hasSeat(int count) {
    return seat >= count;
  }

  void reserveSeat(int count) {
    if (hasSeat(count))
      seat -= count;
    else
      throw new RuntimeException("no seat");
  }*/
}

File: Swift/week2/Week2/Week2/Example/Screening.swift
//
//  Screening.swift
//  Week2
//
//  Created by tskim on 18/10/2019.
//  Copyright © 2019 assin. All rights reserved.
//

import Foundation

class Screening: MemoryHashable {
    private var seat: Int
    let sequence: Int
    let whenScreened: Date

    public init(sequence: Int, whenScreened: Date, seat: Int) {
        self.seat = seat
        self.sequence = sequence
        self.whenScreened = whenScreened
    }
    
    func hasSeat(count: Int) -> Bool {
        return seat >= count
    }
    
    func reserveSeat(count: Int) {
        if hasSeat(count: count) {
            seat -= count
        }
        else {
            fatalError("no seat")
        }
    }
}
//
//extension Screening: Hashable {
//    static func == (lhs: Screening, rhs: Screening) -> Bool {
//        return lhs.hashValue == rhs.hashValue
//    }
//    func hash(into hasher: inout Hasher) {
//        hasher.combine(self.seat)
//        hasher.combine(self.sequence)
//        hasher.combine(self.whenScreened)
//    }
//
//}
